vertx-mysql-client/src/main/java/io/vertx/mysqlclient/MySQLChangeUserOptions.java

```diff
@@ -0,0 +1,249 @@
+   * Set the collation for the connection.
+   *
+   * @param collation the collation to set
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setCollation(String collation) {
+    if (collation != null && !MySQLCollation.SUPPORTED_COLLATION_NAMES.contains(collation)) {
+      throw new IllegalArgumentException("Unsupported collation: " + collation);
+    }
+    this.collation = collation;
+    return this;
+  }
+
+  /**
+   * Get the charset for the connection.
+   *
+   * @return the MySQL collation
+   */
+  public String getCharset() {
+    return charset;
+  }
+
+  /**
+   * Set the charset for the connection.
+   *
+   * @param charset the charset to set
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setCharset(String charset) {
+    if (charset != null && !MySQLCollation.SUPPORTED_CHARSET_NAMES.contains(charset)) {
+      throw new IllegalArgumentException("Unsupported charset: " + charset);
+    }
+    this.charset = charset;
+    return this;
+  }
+
+  /**
+   * Get the path of the server RSA public key.
+   *
+   * @return a reference to this, so the API can be used fluently
+   */
+  public String getServerRsaPublicKeyPath() {
+    return serverRsaPublicKeyPath;
+  }
+
+  /**
+   * Set the path of server RSA public key which is mostly used for encrypting password under insecure connections when performing authentication.
+   *
+   * @param serverRsaPublicKeyPath the path of the server RSA public key
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setServerRsaPublicKeyPath(String serverRsaPublicKeyPath) {
+    this.serverRsaPublicKeyPath = serverRsaPublicKeyPath;
+    return this;
+  }
+
+  /**
+   * Get the value of the server RSA public key.
+   *
+   * @return a reference to this, so the API can be used fluently
+   */
+  public Buffer getServerRsaPublicKeyValue() {
+    return serverRsaPublicKeyValue;
+  }
+
+  /**
+   * Set the value of server RSA public key which is mostly used for encrypting password under insecure connections when performing authentication.
+   *
+   * @param serverRsaPublicKeyValue the path of the server RSA public key
+   * @return a reference to this, so the API can be used fluently
+   */
+  public MySQLChangeUserOptions setServerRsaPublicKeyValue(Buffer serverRsaPublicKeyValue) {
+    this.serverRsaPublicKeyValue = serverRsaPublicKeyValue;
+    return this;
+  }
+
+  public JsonObject toJson() {
+    JsonObject json = new JsonObject();
+    MySQLChangeUserOptionsConverter.toJson(this, json);
+    return json;
+  }
+
+  private void init() {
+    this.user = DEFAULT_USER;
+    this.password = DEFAULT_PASSWORD;
+    this.database = DEFAULT_SCHEMA;
+    this.charset = DEFAULT_CHARSET;
+    this.properties = new HashMap<>(DEFAULT_CONNECTION_ATTRIBUTES);
+  }
+
+  private void checkParameterNonNull(Object parameter, String message) {
```

Comment:
use instead `Objects#requireNonNull` method, it throws `NullPointerException` and this is the correct exception we should throw

Reply:
updated in https://github.com/eclipse-vertx/vertx-sql-client/pull/420/commits/b4967b2c93f2ba463d1106f30a97edf06a25cd26.